treeherder/push_health/push_health.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+from collections import defaultdict
+
+from django.db import connection
+
+from treeherder.model.models import (FailureLine,
+                                     OptionCollection,
+                                     Repository)
+from treeherder.push_health.classification import set_classifications
+from treeherder.push_health.filter import filter_failure
+from treeherder.push_health.utils import clean_test
+
+intermittent_history_days = 14
+intermittent_history_query = '''
+    SELECT fl.test, o.name as 'config', mp.platform
+    FROM treeherder.failure_line as fl
+    LEFT JOIN treeherder.job as j ON fl.job_guid = j.guid
+    LEFT JOIN treeherder.push as p ON j.push_id = p.id
+    LEFT JOIN treeherder.job_type as jt ON j.job_type_id = jt.id
+    LEFT JOIN treeherder.machine_platform as mp ON j.machine_platform_id = mp.id
+    LEFT JOIN treeherder.option_collection as oc ON oc.option_collection_hash = j.option_collection_hash
+    LEFT JOIN treeherder.option as o ON o.id = oc.option_id
+    WHERE fl.repository_id IN (%s)
+    AND p.time >= DATE_SUB(%s, INTERVAL %s DAY)
+    AND j.result = 'testfailed' AND j.tier = 1
+    AND j.failure_classification_id = 4
+    AND fl.test is not null;
+'''
```

Comment:
Have you tried writing this using the Queryset API? I think it'd be doable. My sense of Django is that there's a big emphasis in using the ORM rather than writing raw SQL except for special circumstances. :)

Reply:
I **had** tried, yeah.  But gave up in frustration.  I think my ORM-fu was feeling very rusty.  :)  But you were totally right.  I gave it another shot and ends up this ORM query is actually quite a bit faster.  I'll admit I'm very surprised, but I suppose those folks at Django really know what they're doing.  :)  I wish we were on Postgres, because there are a few features that would become available to us on that engine.

Thank you for your nudge!  :)

Reply:
Awesome! Yeah, it takes some time to dig into the docs but you can do quite a lot with the ORM.